### PR TITLE
Adding more unit tests for ChannelHolder class

### PR DESCRIPTION
**Pull request by abhinavarora**

### CODE REVIEW

File: paddle/fluid/framework/channel_test.cc [hunk 1]
```diff
@@ -542,3 +542,341 @@ TEST(ChannelHolder, ChannelHolderUnBufferedSendReceiveTest) {
...
+  ChannelHolder *ch = new ChannelHolder();
+  ch->Reset<int>(2);
+  EXPECT_EQ(ch->IsInitialized(), true);
+  // Channel should remain intialized even after close
+  ch->close();
+  EXPECT_EQ(ch->IsInitialized(), true);
+  delete ch;
+}
+
+TEST(ChannelHolder, TypeMismatchSendTest) {
```
Comment (chengduoZH) [inline paddle/fluid/framework/channel_test.cc]: Checking type matching or not isn't the function of Channel, is it necessary to add the unit test?

Comment (abhinavarora) [reply paddle/fluid/framework/channel_test.cc]: Channel enforces the datatype in send (https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/fluid/framework/channel.h#L74) and receive (https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/fluid/framework/channel.h#L81)
 
The channel makes sure that the variable being sent or received is same as the type of the channel. The unit test is to make sure the paddle enforce is executed.

Comment (chengduoZH) [reply paddle/fluid/framework/channel_test.cc]: I see, thanks!

Comment (chengduoZH) [reply paddle/fluid/framework/channel_test.cc]: I see, thanks!

**Pull request merged**